File: py/URI_RDF.py
"""
===============================================================================
Creator : Ceri Binding, University of South Wales | Prifysgol de Cymru 
Contact : ceri.binding@southwales.ac.uk
Project : Any
Classes : URI_RDF
Summary : Case-sensitive entity & property URI constants (for RDF apps) 
Example : 
    import URI_RDF
    print(URI_RDF.TYPE) # " http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
License : https://mit-license.org
History :
23/02/2023 CFB Extracted from old script, updated to use F-strings
===============================================================================
"""
# Namespace
NS = "http://www.w3.org/1999/02/22-rdf-syntax-ns#"

# Entities
XMLLITERAL = f"{NS}XMLLiteral"
PROPERTY = f"{NS}Property"
STATEMENT = f"{NS}Statement"
BAG = f"{NS}Bag"
SEQ = f"{NS}Seq"
ALT = f"{NS}Alt"
LIST = f"{NS}List"

# Properties
TYPE = f"{NS}type"
FIRST = f"{NS}first"
REST = f"{NS}rest"
VALUE = f"{NS}value"
SUBJECT = f"{NS}subject"
PREDICATE = f"{NS}predicate"
OBJECT = f"{NS}object"
ID = f"{NS}ID"

File: js/URI_RDF.js
/*
===============================================================================
Creator : Ceri Binding, University of South Wales | Prifysgol de Cymru 
Contact : ceri.binding@southwales.ac.uk
Project : Any
Classes : URI_RDF 
Summary : Case-sensitive entity & property URI constants (for RDF apps) 
Example : 
    <script type="module">
        import {URI_RDF} from "./URI_RDF.js"
        console.log(URI_RDF.BAG) | console.log(URI_RDF["BAG"])
        # both return "http://www.w3.org/1999/02/22-rdf-syntax-ns#Bag"
    </script>
License : https://mit-license.org
History :
10/02/2023 CFB Extracted from legacy script and converted to ES6 module
===============================================================================
*/
const NS = "http://www.w3.org/1999/02/22-rdf-syntax-ns#"
export const URI_RDF = Object.freeze({
    // entities
    ALT: `${NS}Alt`,
    BAG: `${NS}Bag`,
    LIST: `${NS}List`,
    PROPERTY: `${NS}Property`,
    SEQ: `${NS}Seq`,
    STATEMENT: `${NS}Statement`,
    XMLLITERAL: `${NS}XMLLiteral`,
    // properties
    FIRST: `${NS}first`,
    ID: `${NS}ID`,
    OBJECT: `${NS}object`,
    PREDICATE: `${NS}predicate`,
    REST: `${NS}rest`,
    SUBJECT: `${NS}subject`,
    TYPE: `${NS}type`,
    VALUE: `${NS}value`
})